lib/oga/entity_decoder.rb

```diff
@@ -13,6 +13,7 @@ def self.try_decode(input, html = false)
     # @return [String]
     #
     def self.decode(input, html = false)
+      fail ArgumentError, "decode was sent nil" if input.nil?
```

Comment:
It's `fail` because it's a failure, not a re-raise. I'm not raising the error, it's failing a contract.

`try_decode` does check for this (though I couldn't see a reason why it's structured this way, a comment would be nice) but that's the point - if it gets this fair it should fail. Hence why I didn't add anything to `try_decode`, even though it should probably have more argument checks.